### PR TITLE
core-image-pelux: turn recipe into a class

### PR DESCRIPTION
**Pull request by mardy**
Modify meta-pelux to make the "core-image-pelux" recipe into a bbclass,
rather than a recipe.
Having core-image-pelux be a bbclass rather than a recipe will allow us
to use the "inherit" directive in our image files, rather than the
"require" directive. "inherit" is preferred, since it does not require
us to specify the path to the bbclass.

Update the existing recipes to work with this change.

### CODE REVIEW

**Comment by GordanM** [comment]
@zeenix care to comment on this commit message? :)

**Comment by GordanM** [comment]
Fixes #34 .

File: classes/core-image-pelux.bbclass [hunk 1]
```diff
@@ -40,3 +38,4 @@ toolchain_create_sdk_env_script_append() {
 IMAGE_ROOTFS_SIZE ?= "1000000"
 
 IMAGE_FSTYPES ?= "ext3 sdcard"
+
```
Comment (zeenix) [inline classes/core-image-pelux.bbclass]: Unrelated change, please remove (from this patch).

File: classes/core-image-pelux.bbclass [hunk 2]
```diff
@@ -3,8 +3,6 @@
 #   SPDX-License-Identifier: MIT
 #
 
-DESCRIPTION = "Image for creating a Pelux image"
-
```
Comment (zeenix) [inline classes/core-image-pelux.bbclass]: Unrelated change, please remove (from this patch).

**Comment by zeenix** [comment]
@GordanM I guess you meant that Alberto should add `Fixes #34` to the commit log?

File: classes/core-image-pelux.bbclass (hunk 2 above)
Comment (mardy) [reply classes/core-image-pelux.bbclass]: This one is not unrelated. The diff is misleading because git detects the addition of this new file as a move from the core-image-pelux.bb, since the contents are indeed mostly the same.

But in fact, I didn't remove these lines: simply didn't copy them into the new file, as they should not be added there.

Comment (zeenix) [reply classes/core-image-pelux.bbclass]: ah, the move wasn't obvious to me.

**Comment by GordanM** [comment]
@zeenix the `Fixes #34` is not needed in the commit log. I've just linked the issues by stating #34 here in the comments.

**Comment by zeenix** [comment]
didn't we decide on having some sort of trace in the commit logs to issues?

**Comment by mardy** [comment]
I update che branch. I added "Fixes #34" in the commit message too, I also think it's a good to have it there.

**Comment by GordanM** [comment]
@breakreturn can you please check the comments above regarding the issue tracing?

**Comment by breakreturn** [comment]
@GordanM Most discussions that have been difficult to resolve on this topic have been when the situation involves an internal issue tracker and referencing that in the commit. That's not something I'd say we should do.

In this case, where we use the open issue tracker I don't particularly mind referencing it. However, I do not think we have any solid strategy or plan wrt exactly how to use the open issue tracker or reference it so I can't say what the consensus is here. This is fine for me as it is.

**Comment by GordanM** [comment]
Thanks @breakreturn, essentially that is the same thing I've told to @zeenix IRL: The majority of our backlog is found in an internal issue tracker and that is the reason why I find having references to github issues slightly redundant, as there is no common pattern.
